### ISSUE TITLE
Contact Method -  For the same account, changing the default contact method on one device is not always dynamically applied to other devices for

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#22163]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
PRECONDITION: Account A has many contact methods added
1. Log in to account A on "Device 1" and "Device 2"
2. On the "Device 1", go to “Profile settings” >> select other contact method and “Set as default”
3. Check the contact method information of Account A on "Device 2"

## Expected Result:
Changing the default contact method on Device 1 should be dynamically applied to Device 2

## Actual Result:
Sometimes, changing the default contact method on Device 1 IS NOT dynamically applied to Device 2 - It may still show previous contact method on Profile information

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.34-1
**Reproducible in staging?:** y
**Reproducible in production?:** y

**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
https://github.com/Expensify/App/assets/93399543/a2832b82-c25f-4fd7-b88f-3fa2d6a54544

https://github.com/Expensify/App/assets/93399543/7697d133-69df-490b-96ef-be664c4b244c

https://github.com/Expensify/App/assets/93399543/fdbd376b-5414-4535-9b43-1d95dad29769

**Expensify/Expensify Issue URL:**
**Issue reported by:** ToanSel
QA Engineer - Github: @tranvantoan-qn
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1688323751728569

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by adelekennedy** [comment]
reassigning - I got hit with a ton of issues right before getting sick

**Comment by sophiepintoraetz** [comment]
Hey @adelekennedy , sorry you're not feeling 100% - happy to help keep this issue moving along! However, because I think you'll (hopefully) only be out of action for a few days - it's best that you keep yourself assigned until you're back in action - as[ per the SO](https://stackoverflowteams.com/c/expensify/questions/8836). I'll take care of this one but keep this in mind for next time, as the team is slammed atm (esp with 4 July outage!)

**Comment by StevenKKC** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
For the same account, changing the default contact method on one device is not always dynamically applied to other devices.

### What is the root cause of that problem?
In Profile page, contact method is displayed by below code.
https://github.com/Expensify/App/blob/5cca3c1436cab81b013c7d1d77660b1595a6fcdf/src/pages/settings/Profile/ProfilePage.js#L70-L75

After user changes the default contact method on one device, BE will push the events that contains updated `session` and `personalDetailsList` to another device.
`session` includes new default contact method, but `personalDetailsList` does not have new contact method.

So the default method does not update dynamically.

### What changes do you think we should make in order to solve the problem?
There are two solutions.
- Modify BE in order to push `personalDetailsList` that includes new default contact method.
- Modify App in order to display new default contact method contained in `session`. 
For this, we add `session` props in `ProfilePage.js`, and add `propTypes` and `defaultProps` for `session`.
```js
export default compose(
    ...
    withOnyx({
        ...
+       session: {
+           key: ONYXKEYS.SESSION,
+       },
    }),
)(ProfilePage);
```
And display new default contact method contained in `session` as below.
```js
-   title: props.formatPhoneNumber(lodashGet(currentUserDetails, 'login', '')),
+   title: props.formatPhoneNumber(props.session.email),
```
### What alternative solutions did you explore? (Optional)
None.

**Comment by sophiepintoraetz** [comment]
Able to reproduce!

https://github.com/Expensify/App/assets/42940078/e7860a51-3179-47cb-9570-a701c600213a
<img width="779" alt="image" src="https://github.com/Expensify/App/assets/42940078/ea11ac6e-1514-4a74-9605-85b9af9a90c8">

**Comment by Santhosh-Sellavel** [comment]
@sophiepintoraetz Sorry for the delay This might require a BE Change should be internal first!

**Comment by sophiepintoraetz** [comment]
@Santhosh-Sellavel - lol, does that mean you can do it...?

**Comment by Santhosh-Sellavel** [comment]
@sophiepintoraetz No I can't. It's added because C+ Reviews internal issues that need handling in E/App.

**Comment by Santhosh-Sellavel** [comment]
Please add engineering label

**Comment by Santhosh-Sellavel** [comment]
Bump @joelbettner

**Comment by joelbettner** [comment]
@Santhosh-Sellavel Can you provide some more detailed description about the back end change that is being requested?

**Comment by joelbettner** [comment]
@Santhosh-Sellavel Ok, I've dug into this quite a bit more.

The `personalDetailsList` should already include the default contact info. It should be the `login` element of the `personalDetailsList` in the Onyx data.

Here is a screenshot of the info in my `personalDetailsList`:
![image](https://github.com/Expensify/App/assets/13264774/7bb8f3ad-d47d-47a6-922a-8b466a6a1075)

Here is a screenshot of the info in my `personalDetailsList` after changing it to a phone number:
![image](https://github.com/Expensify/App/assets/13264774/99619712-7e1b-48b8-8289-02663a9dd4b7)

And here is a screenshot of the info in my `personalDetailsList` after changing it back to my Expensify email again:
![image](https://github.com/Expensify/App/assets/13264774/260d32c3-08ba-413e-8ab9-a588ad9adbcc)

**Comment by joelbettner** [comment]
Also, I've been testing this and I'm unable to re-create the issue.

**Comment by joelbettner** [comment]
I don't think this is actually an issue. I'm going to close. Feel free to re-open if necessary.

**Comment by Santhosh-Sellavel** [comment]
Thanks @joelbettner !

**Comment by Santhosh-Sellavel** [comment]
@tranvantoan-qn This is not occurring at all here is the recording

https://github.com/Expensify/App/assets/85645967/2467a202-694f-4d96-95eb-f95001265801

**Comment by tranvantoan-qn** [comment]
I'm UNABLE to reproduce it now because it seems this function is BLOCKED for me for **2 days** with the error `401 Request rejected - too many attempts`
Do you know how long will I be blocked from using this function?

https://github.com/Expensify/App/assets/10254571/7718f343-9e08-4dd9-83b7-56c0dbca94bc

**Comment by Santhosh-Sellavel** [comment]
There might be API limits.

 Please always use Staging. [This](https://github.com/Expensify/App/issues/22163#issuecomment-1644483357) is from staging. Everything looks fine, thanks!

**Comment by tranvantoan-qn** [comment]
Another question is:
Initially, this is a valid issue report, and BZ could reproduce it too. So, if it is fixed by the backend, is it eligible for a bug reporter bonus? 
🤔

cc @sophiepintoraetz @joelbettner

**Comment by tranvantoan-qn** [comment]
@Santhosh-Sellavel 
Yeah, I did use staging.
